sale_order_project/views/sale_view.xml

```diff
@@ -1,24 +1,33 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
-    <data>
-        <record id="view_order_form" model="ir.ui.view">
-            <field name="name">view_project_quotation_order_form</field>
-            <field name="model">sale.order</field>
-            <field name="inherit_id" ref="sale.view_order_form" />
-            <field name="arch" type="xml">
-                <group name="technical" position="after">
-                    <group name="project" string="Project">
-                        <field name="related_project_id" />
-                    </group>
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sales.order.project.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <group name="technical" position="after">
+                <group name="project" string="Project">
+                    <field name="related_project_id" />
                 </group>
-                <xpath expr="//header/field[@name='state']" position="before">
-                    <button name="action_create_project" type="object"
-                            string="Create Project" class="btn-primary"
-                            groups="project.group_project_user"
-                            attrs="{'invisible':['|',('state','!=','draft'),('related_project_id','!=',False)]}" />
-                </xpath>
-            </field>
-        </record>
-
-    </data>
-</openerp>
+            </group>
+            <xpath expr="//button[@name='action_view_invoice']" position="before">
+                <!-- icon="fa-project-diagram"
+                    This must to be the proper icon, but, until Odoos 12 version the current Font-Awesome version is 4.7, such has not this icon; instead we will use the current icon.
+                    https://github.com/odoo/odoo/issues/21648#issuecomment-353099417
+                -->
+                <button class="oe_stat_button"
+                        type="object"
+                        name="action_create_project"
+                        icon="fa-tasks"
+                        help="Create Project"
+                        groups="sale.group_analytic_accounting,project.group_project_user"
+                        attrs="{'invisible':['|',('state','!=','draft'),('related_project_id','!=',False)]}">
```

Comment:
If you want a confirmation add a new attribute: ``` confirmation="Text to show on the confirm dialog". ```

Reply:
![deepin-screen-recorder_select area_20180806100950](https://user-images.githubusercontent.com/5034215/43702026-1529ab7c-9961-11e8-9c26-9fd9ea60ae79.gif)
I'm thinking about something like this.

Reply:
If you don't want the user to input something (like the name of the project to create) I think that confirmation dialog is enough.

Reply:
I prefer this approach:
> In a manufacturing order (MO), you can select a project to be attached to it. If none is selected, a project is automatically created when the MO is confirmed.
It is the [`mrp_project`](https://github.com/OCA/manufacture/tree/8.0/mrp_project) module.

I mean, if the user wants to attach an existing  project user can choose, but if the user does not choose anything, the current behavior is the ideal.

Reply:
Ok, do you know how to make this?

Reply:
I will check it.

Reply:
@aitorbouzas I'm not sure about how can i create a new wizard, i made the form, but i don't know how to do the other things.

Anyway, i'm searching about that.

Reply:
The button should return an action or call a method that returns an action. This is an example of a button that returns an action:
```
<button class="oe_highlight" attrs="{'invisible': [('sale_number','=', 0)]}" type="action"
                        string="Make certification(s)"
                        name="%(crm_lead_make_invoice_action)d" />
```

And this is the action that shows the popup form:

```
<act_window name="Make invoice from lead" id="crm_lead_make_invoice_action"
                res_model="crm.lead.make_invoice"
                view_mode="form" target="new"/>
```

You also have to have a form view for the wizard model ```crm.lead.make_invoice``` that will define the wizard view.

Reply:
Yes, i did all of it, but the `python` code is not so easy, i think.

Reply:
@aitorbouzas Can you review my recent commit? please.